lib/logstash/codecs/compress_spooler.rb

```diff
@@ -21,13 +21,13 @@ def decode(data)
   end # def decode
 
   def encode(event)
+    # use normalize to make sure returned Hash is pure Ruby for
+    # MessagePack#pack which relies on pure Ruby object recognition
+    @buffer << LogStash::Util.normalize(event.to_hash).merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)
+    # If necessary, we flush the buffer and get the data compressed
```

Comment:
Just checking, with normalize and event.to_hash there is a good chance that one is pushing a pointer to the events `@data` ivar, but by using merge on it one makes a copy and that is pushed to the buffer. I think event.to_hash should make a shallow copy.